subsys/bluetooth/host/conn.c

```diff
@@ -402,13 +404,19 @@ static void bt_acl_recv(struct bt_conn *conn, struct net_buf *buf,
 		/* Still not enough data received to retrieve the L2CAP header
 		 * length field.
 		 */
+		bt_send_one_host_num_completed_packets(conn->handle);
+		*buf_flags = ACL_DATA_HOST_NUM_COMPLETE_SENT;
+
 		return;
 	}
 
 	acl_total_len = sys_get_le16(conn->rx->data) + sizeof(struct bt_l2cap_hdr);
 
 	if (conn->rx->len < acl_total_len) {
 		/* L2CAP frame not complete. */
+		bt_send_one_host_num_completed_packets(conn->handle);
+		*buf_flags = ACL_DATA_HOST_NUM_COMPLETE_SENT;
```

Comment:
Since it seems that `bt_send_one_host_num_completed_packets` can fail, should we really onconditionally set `*buf_flags = ACL_DATA_HOST_NUM_COMPLETE_SENT;`?

Reply:
I think the host's broken if that's the case. It allocates with `K_FOREVER` unless syswq. 
If we are in the syswq (likely) and there are no command buffers, then I guess we could schedule a work item to retry later. If it's the command send that failed, then we can't do anything about it.

I can make a small change, but I don't think it will have an impact:
- return an int based on what step failed: alloc vs send
- don't set the bit if alloc failed
- `net_buf_unref()` will trigger the alloc+send again

I don't think it will do anything because the unref is done from the same context a line below. So if there are no free command buffers and we are running in a coop thread, the result will be the same.

Reply:
When can this happen? 
And if we cannot recover from it, isn't that effectively a fatal error. Does it warrant an assert?

Reply:
Yep. I'll add a BT_ASSERT, just like when send_sync fails